aws/kinesis/core/pipeline.h

```diff
@@ -190,8 +200,34 @@ class Pipeline : boost::noncopyable {
     });
   }
 
-  std::string stream_;
+  // Retrieve the account ID and partition from the STS service.
+  static std::string init_stream_arn(const std::string &region, const std::string &stream_name) {
+    Aws::STS::STSClient sts;
+    Aws::STS::Model::GetCallerIdentityRequest request;
+    auto outcome = sts.GetCallerIdentity(request);
+    if (outcome.IsSuccess()) {
+      auto result = outcome.GetResult();
+      Aws::Utils::ARN sts_arn(result.GetArn());
+
+      // Construct and return the Kinesis stream ARN.
+      std::stringstream arn;
+      arn << "arn:" << sts_arn.GetPartition() << ":kinesis:" << region << ":" << result.GetAccount()
+          << ":stream/" << stream_name;
+
+      auto arn_str = arn.str();
+      LOG(info) << "StreamARN \"" << arn_str << "\" has been successfully configured, "
+                << "and will be used in requests including ListShards and PutRecords";
+      return arn_str;
+    }
+
+    LOG(warning) << "Failed to get StreamARN using STS GetCallerIdentity with exception: "
```

Comment:
we should throw if this fails. 

that will avoid operating in dual-mode, and have all applications start using ARN from this version onwards.